### PR TITLE
JDK-8294748: Cleanup unneeded references to hg

### PR DESCRIPTION
**Pull request by MBaesken**
There are still a number of unneeded references to hg, for example

make/autoconf/spec.gmk.in
 
772 HG:=@HG@
 
make/autoconf/basic_tools.m4
 
345 UTIL_LOOKUP_PROGS(HG, hg)

Those can be cleaned up.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294748](https://bugs.openjdk.org/browse/JDK-8294748): Cleanup unneeded references to hg

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10554/head:pull/10554` \
`$ git checkout pull/10554`

Update a local copy of the PR: \
`$ git checkout pull/10554` \
`$ git pull https://git.openjdk.org/jdk pull/10554/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10554`

View PR using the GUI difftool: \
`$ git pr show -t 10554`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10554.diff">https://git.openjdk.org/jdk/pull/10554.diff</a>

</details>

### CODE REVIEW

File: make/common/MakeBase.gmk
Comment (erikj79) [inline make/common/MakeBase.gmk:133]: This line should be possible to remove as well.

Comment (MBaesken) [reply make/common/MakeBase.gmk]: Hi Erik, thanks for the remark. I removed the line.

**Review by erikj79 — APPROVED** [review]

**Comment by MBaesken** [comment]
Hi Erik, thanks for approving ;  do we need a second review or is it sufficient now ?
Best regards, Matthias

**Review by RealCLanger — APPROVED** [review]
Looks good. Thanks for the cleanup.

**Pull request closed without merging**
